src/name/mlopatkin/andlogview/liblogcat/file/FileDataSourceFactory.java

```diff
@@ -71,15 +73,15 @@ public static ImportResult createDataSource(String fileName, CharSource file)
     }
 
     private static ImportResult createLogFileSource(String fileName, LogcatFormatSniffer formatSniffer,
-            ReplayParser<?> replayParser, BufferedReader in)
+            ReplayParser<?> replayParser, LineReader in)
             throws IOException {
         return new LogfileDataSource.Builder(fileName).setParserFactory(
                         handler -> FormatSniffer.createAndReplay(replayParser, formatSniffer::createParser, handler))
                 .readFrom(in);
     }
 
     private static ImportResult createDumpstateFileSource(String fileName, DumpstateFormatSniffer formatSniffer,
-            ReplayParser<?> replayParser, BufferedReader in) throws IOException, UnrecognizedFormatException {
+            ReplayParser<?> replayParser, LineReader in) throws IOException, UnrecognizedFormatException {
         return new DumpstateFileDataSource.Builder(fileName).setParserFactory(
                 h -> FormatSniffer.createAndReplay(replayParser, formatSniffer::createParser, h)).readFrom(in);
```

Comment:
Apparently, there we feed EOLN-ending lines into dumpstate parser when replaying.